File: app/src/main/java/fr/mpau/tools/lists/InterventionsAdapter.java
package fr.mpau.tools.lists;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

import fr.mpau.R;
import fr.mpau.enums.SubtypeIntervention;
import fr.mpau.enums.TypeIntervention;
import fr.mpau.models.Intervention;
import fr.mpau.tools.StringTools;

/**
 * Adapter pour afficher la ListView des interventions
 * <p>
 * Author: Jonathan B.
 * Created: 19/01/2018
 */

public class InterventionsAdapter extends ArrayAdapter<Intervention> {

    /**
     * Attributs
     */
    private final Context context;
    private ArrayList<Intervention> listInterventions;

    /**
     * Constructeur
     *
     * @param context           Context
     * @param resource          int
     * @param listInterventions ArrayList<Intervention>
     */
    public InterventionsAdapter(Context context, int resource, ArrayList<Intervention> listInterventions) {
        super(context, resource, listInterventions);
        this.context = context;
        this.listInterventions = listInterventions;
    }

    /**
     * Construction d'une entrée (une ligne d'intervention)
     *
     * @param position    int
     * @param convertView View
     * @param parent      ViewGroup
     * @return View
     */
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            if (inflater != null) {
                convertView = inflater.inflate(R.layout.row_intervention, parent, false);
            }
        } else {
            convertView = (RelativeLayout) convertView;
        }
        // Récupération des éléments du layout
        TextView typeInter = convertView.findViewById(R.id.typeInter);
        TextView subtypeInter = convertView.findViewById(R.id.subtypeInter);
        TextView smurInter = convertView.findViewById(R.id.smurInter);
        TextView secteurInter = convertView.findViewById(R.id.secteurInter);
        TextView commentInter = convertView.findViewById(R.id.commentInter);
        TextView datetimeInter = convertView.findViewById(R.id.datetimeInter);
        // Récupération de l'item Intervention
        Intervention inter = listInterventions.get(position);
        // Inflate le layout
        // Affichage 'SMUR'
        if (inter.isInterSmur()) {
            String smurText = "   " + context.getResources().getString(R.string.smur_tag);
            smurInter.setText(smurText);
            smurInter.setVisibility(View.VISIBLE);
        } else {
            smurInter.setVisibility(View.GONE);
        }
        // Formattage de l'heure
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(inter.getInterDate() * 1000);
        Date date = calendar.getTime();
        SimpleDateFormat formatDateTime = new SimpleDateFormat("dd/MM/yyyy  HH:mm", Locale.getDefault());
        typeInter.setText(TypeIntervention.values()[inter.getInterTypeId() - 1].toString());
        String subType = " - " + SubtypeIntervention.values()[inter.getInterSoustypeId() - 1].toString();
        subtypeInter.setText(subType);
        secteurInter.setText(inter.getInterSecteur().trim());
        // Formattage du commentaire
        String rawComment = inter.getInterCommentaire();
        String comment = StringTools.formatComment(rawComment);
        commentInter.setText(comment);
        String dateTime = formatDateTime.format(date) + "\n";
        datetimeInter.setText(dateTime);
        return convertView;
    }
}

File: src/main/java/fr/mpau/tool/list/InterventionsAdapter.kt
package fr.mpau.tool.list

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.RelativeLayout
import android.widget.TextView
import fr.mpau.R
import fr.mpau.bean.Intervention
import fr.mpau.enums.AgePatientIntervention
import fr.mpau.enums.SubtypeIntervention
import fr.mpau.enums.TypeIntervention
import fr.mpau.tool.StringTools
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale

/**
 * Tool - Adaptateur pour afficher la ListView des interventions
 *
 * Author: Jonathan
 * Created: 19/01/2018
 * Last Updated: 27/01/2025
 */
class InterventionsAdapter(
    private val context: Context,
    resource: Int,
    list: ArrayList<Intervention>
) : ArrayAdapter<Intervention?>(context, resource, list as List<Intervention?>) {

    /**
     * Attributs
     */

    private val listInterventions: ArrayList<Intervention> = list

    /**
     * Construction d'une entrée (une ligne d'intervention)
     *
     * @param position Int
     * @param convertView View?
     * @param parent ViewGroup
     * @return View
     */
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            val inflater: LayoutInflater =
                context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.row_intervention, parent, false)
        } else {
            view = view as RelativeLayout
        }
        // Récupération des éléments du layout
        val mainTypeInter = view!!.findViewById<TextView>(R.id.mainTypeInter)
        val subTypeInter = view.findViewById<TextView>(R.id.subTypeInter)
        val smurInter = view.findViewById<TextView>(R.id.smurInter)
        val agePatientInter = view.findViewById<TextView>(R.id.agePatientInter)
        val sectorInter = view.findViewById<TextView>(R.id.sectorInter)
        val commentInter = view.findViewById<TextView>(R.id.commentInter)
        val datetimeInter = view.findViewById<TextView>(R.id.datetimeInter)
        val durationInter = view.findViewById<TextView>(R.id.durationInter)
        // Récupération de l'item Intervention
        val inter: Intervention = listInterventions[position]
        // Inflation des éléments
        val mainType = TypeIntervention.entries[inter.mainTypeId - 1].getValue()
        mainTypeInter.text = mainType
        val subType = " - " + SubtypeIntervention.entries[inter.sousTypeId - 1].getValue()
        subTypeInter.text = subType
        if (inter.smur) {
            val smur = "   " + context.resources.getString(R.string.smur_tag)
            smurInter.text = smur
            smurInter.visibility = View.VISIBLE
        } else {
            smurInter.visibility = View.GONE
        }
        agePatientInter.text = AgePatientIntervention.entries[inter.agePatientId - 1].getValue()
        val sector = " ans - " + inter.secteur.trim()
        sectorInter.text = sector
        // Formattage du commentaire
        val rawComment: String = inter.commentaire
        val comment: String = StringTools.formatComment(rawComment)
        commentInter.text = comment
        // Formattage de l'heure
        val calendar = Calendar.getInstance()
        calendar.timeInMillis = inter.dateInter * 1000
        val date = calendar.time
        val formatDateTime = SimpleDateFormat("dd/MM/yyyy  HH:mm", Locale.getDefault())
        val dateTime = formatDateTime.format(date) + "\n"
        datetimeInter.text = dateTime
        val duration = "  " + inter.duree.toString() + " min"
        durationInter.text = duration
        return view
    }

}